core/src/main/java/org/jdbi/v3/core/rewriter/ParsedParameters.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.rewriter;
+
+import java.util.List;
+import java.util.Objects;
+
+public class ParsedParameters {
+    private final boolean positional;
+    private final List<String> parameterNames;
+
+    ParsedParameters(boolean positional, List<String> parameterNames) {
+        this.positional = positional;
+        this.parameterNames = parameterNames;
```

Comment:
Should there be an `unmodifiableList` wrapper here or some other mutation protection?